src/software/networking/proto_multicast_sender.tpp

```diff
@@ -10,20 +10,32 @@ ProtoMulticastSender<SendProto>::ProtoMulticastSender(boost::asio::io_service& i
 {
     boost::asio::ip::address multicast_addr = boost::asio::ip::make_address(ip_address);
 
+    // Create the sender_endpoint. This is the endpoint that identifies the sender, and
+    // must be different than the receiver (otherwise multiple sockets will try bind to
+    // the same port and cause errors). Port 0 is a special value that tells the operating
+    // system to dynamically assign an available port.
+    // https://www.lifewire.com/port-0-in-tcp-and-udp-818145
+    boost::asio::ip::udp::endpoint sender_endpoint =
+        boost::asio::ip::udp::endpoint(multicast_addr, 0);
+    ;
+    // The receiver endpoint identifies where this MulticastSender will send data to
```

Comment:
hmm I think this might not even be necessary, would you be able to remove this and the `bind` call altogether

`send_to` should handle it internally

Reply:
Just to be clear, you mean removing the `sender_enpoint`?

I removed the `sender_endpoint` and `bind` call and everything works.

Reply:
yup thats it